sources/iTermPreferences.m

```diff
@@ -279,9 +279,13 @@ + (BOOL)defaultValueForKey:(NSString *)key isCompatibleWithType:(PreferenceInfoT
     id defaultValue = [self defaultValueMap][key];
     switch (type) {
         case kPreferenceInfoTypeIntegerTextField:
+        case kPreferenceInfoTypeUnsignedIntegerTextField:
         case kPreferenceInfoTypePopup:
             return ([defaultValue isKindOfClass:[NSNumber class]] &&
                     [defaultValue doubleValue] == ceil([defaultValue doubleValue]));
+        case kPreferenceInfoTypeUPopup:
+            return ([defaultValue isKindOfClass:[NSNumber class]] &&
+                    [defaultValue unsignedLongValue] == ceil([defaultValue unsignedLongValue]));
```

Comment:
Use `unsignedIntegerValue`, not `unsignedLongValue` for consistency. Doesn't matter in practice as both are 64 bit.